Source/WebCore/css/StyleRule.cpp

```diff
@@ -319,42 +300,30 @@ Vector<RefPtr<StyleRule>> StyleRule::splitIntoMultipleRulesWithMaximumSelectorCo
     return rules;
 }
 
-StyleRuleRareData& StyleRule::rareData() const
-{
-    if (!m_rareData)
-        m_rareData = makeUnique<StyleRuleRareData>();
-    return *m_rareData;
-}
 
-void StyleRule::setNestedRules(Vector<Ref<StyleRuleBase>> nestedRules)
+const CSSSelectorList& StyleRule::resolvedSelectorList() const
 {
-    if (!m_rareData && nestedRules.isEmpty())
-        return;
-    rareData().nestedRules = WTFMove(nestedRules);
+    if (isStyleWithNestingRule())
+        return downcast<StyleWithNestingRule>(*this).resolvedSelectorList();
+
+    return m_selectorList;
 }
 
-void StyleRule::setResolvedSelectorList(CSSSelectorList&& resolvedSelectorList) const
-{
-    if (!m_rareData && resolvedSelectorList.isEmpty())
-        return;
-    rareData().resolvedSelectorList = WTFMove(resolvedSelectorList);
+Ref<StyleWithNestingRule> StyleWithNestingRule::create(Ref<StyleProperties>&& properties, bool hasDocumentSecurityOrigin, CSSSelectorList&& selectors, Vector<Ref<StyleRuleBase>>&& nestedRules)
+{ 
+    return adoptRef(* new StyleWithNestingRule(WTFMove(properties), hasDocumentSecurityOrigin, WTFMove(selectors), WTFMove(nestedRules)));
 }
 
-static const Vector<Ref<StyleRuleBase>>& emptyRuleVector()
-{
-    static LazyNeverDestroyed<Vector<Ref<StyleRuleBase>>> emptyRuleVector;
-    static std::once_flag onceFlag;
-    std::call_once(onceFlag, [] {
-        emptyRuleVector.construct();
-    });
-    return emptyRuleVector;
+StyleWithNestingRule::StyleWithNestingRule(Ref<StyleProperties>&& properties, bool hasDocumentSecurityOrigin, CSSSelectorList&& selectors, Vector<Ref<StyleRuleBase>>&& nestedRules)
```

Comment:
`StyleRuleWithNesting` would be more consistent with the existing naming scheme here.

Reply:
Isn't the naming scheme `XXXXXRule` ?

Reply:
Well there is `StyleRuleGroup` so clearly it's not always `XXXRule` ... (but maybe we should remove StyleRuleGroup for the hierarchy actually)

Reply:
the scheme here is always start with StyleRule:

StyleRuleFontFace
StyleRuleFontPaletteValues
StyleRulePage
StyleRuleGroup
StyleRuleMedia
StyleRuleProperty
....

Reply:
Oh yes it's the CSS ones which are `CSSXXXRule`

Reply:
Changed to `StyleRuleWithNesting`